src/patch.rkt

```diff
@@ -98,6 +102,36 @@
           (sow (alt expr* `(taylor ,name ,var (2)) (list altn)))))
       (timeline-stop!))))
 
+
+(define taylor-sample
+  (parameterize ([bf-precision 53])
+    (map (lambda (num) (ival num))
+         (list (bf 0.0) (bf 0.5) (bf -0.5)
+               (bfprev +inf.bf)
+               (bfnext -inf.bf)))))
+
+(define (filter-valid alts)
```

Comment:
Ok, now that I've read it, this is pretty weird. Do I get it right, that you're evaluating the taylor-expanded program on `(x x x x ... x)` where `x` is one of 0, -1/2, 1/2, +inf, or -inf? And passing it if it's one of these? And you're evaluating in the reals, not in the floats?

Why not evaluate on all sampled points? (Too slow?) Why not evaluate in the floats? What kinds of programs are we actually trying to filter out here—can we come up with a better predicate for "invalid on all points", like maybe evaluating on a single `[-inf, +inf]` interval?

Reply:
You've got it right.

> Why not evaluate on all sampled points? (Too slow?)

All the sampled points is a bit slow. 

> Why not evaluate in the floats?

Floats might miss errors in the reals. Imagine
`(if complex_float_comutation_that_rounds (/ 1 0) 1)`
If the float computation rounds the right way it may miss that  `(/ 1 0)` is invalid for that point

> What kinds of programs are we trying to filter out?

We want to filter out programs that are invalid on all points. It's taylor's goal to make the program more accurate on some specific part of the input (0, inf, -inf) so we probably don't care if it's invalid outside of that.

If we made a version of taylor that was valid on all points that would be cool, but when I tried filtering for all points it hurt accuracy.

> can we come up with a better predicate for "invalid on all points", like maybe evaluating on a single [-inf, +inf] interval?

I thought about a singe interval, but I think it would also throw away useful programs due to the overapproximation

Reply:
Hmm, ok, if we evaluate on a single `[-inf, +inf]` interval, we'd get one of three error intervals: true (meaning valid on all inputs), false (meaning invalid on all inputs), and uncertain (meaning possibly invalid on some or all or none). What if we only filter out programs that return "false"? Those are _definitely_ safe to filter out. Some always-invalid programs will make it through (like `sqrt(x) sqrt(-x) / x`) but that seems rare?

I'm hammering on this point because the condition here is pretty weird (we got this crazy list of five points?) and I think the goals we're trying to achieve (filter out things with effectively NaNs in them) pretty simple.

Reply:
I'm starting to re-think this whole thing. My latest version of herbie on egglog can handle programs with errors in them, including if they error on the entire domain

Reply:
So we no longer need to make it sound for egglog's sake.
So is there any benefit to this PR or line of thinking?

Reply:
Hmm, ok, I'd love to hear more about egglog progress in a later meeting. (How come we don't need validity?)

The real goal of this branch was to start merging the "easy" parts of the egglog branch. If taylor sound isn't necessary, maybe we just drop the PR then?